### PR TITLE
Add tutorial to documentation

### PR DESCRIPTION
**Pull request by bvreede**
Current process:
- export the notebook to RST
- move it to docs folder

Problem is the layout of the output cells (i.e. no visual difference with code cells).

We found [nbsphinx](https://nbsphinx.readthedocs.io/) which may help? Currently exploring.

### CODE REVIEW

**Comment by Peter9192** [comment]
nbsphinx works for me after:

- Add "nbsphinx" to `extensions` in `docs/conf.py`
- `pip install nbsphinx` (need to add to package)
- copy or move the desired notebook (just the ipynb file) to the docs folder (I tried with `examples/Irrigation.py`)
- add `Irrigation` (without `.ipynb`) to `docs/index.rst` under 
- `python setup.py build_sphinx -Ea`
- Inspect rendered docs looks really beautiful!

This is so nice I suggest moving our entire example notebooks directory straight into the docs :-)

**Comment by bvreede** [comment]
Thanks for taking a look, @SarahAlidoost! 

This PR moves all the notebooks into the documentation, so they can be viewed as part of the package's readthedocs. [Check the rendered documentation here](https://ewatercycle--117.org.readthedocs.build/en/117/), or to run it locally:

- run `pip install -e .[dev]` again (some more dependencies were added) — or simply run `pip install nbsphinx` in the conda environment
- render the documentation with `python setup.py build_sphinx -Ea`

The notebooks need to be cleaned up, but that is outside the scope of this PR :)

Something to check is the order of the notebooks in the index, and whether the title is good.

File: docs/examples.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
+Examples
+========
+
+.. toctree::
+
+  Platform tutorial <examples/ewatercycle_api_notebook.ipynb>
+  Pre-processing input <examples/preprocessing.ipynb>
```
Comment (SarahAlidoost) [inline docs/examples.rst]:
```suggestion
  Pre-processing <examples/preprocessing.ipynb>
```

File: docs/examples.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  Lisflood <examples/lisflood.ipynb>
```
Comment (SarahAlidoost) [inline docs/examples.rst]:
```suggestion
  LISFLOOD <examples/lisflood.ipynb>
```

File: docs/examples.rst [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  Marrmot M01 <examples/MarrmotM01.ipynb>
```
Comment (SarahAlidoost) [inline docs/examples.rst]:
```suggestion
  MARRMoT M01 <examples/MarrmotM01.ipynb>
```

File: docs/examples.rst [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  Marrmot M14 <examples/MarrmotM14.ipynb>
```
Comment (SarahAlidoost) [inline docs/examples.rst]:
```suggestion
  MARRMoT M14 <examples/MarrmotM14.ipynb>
```

**Review by SarahAlidoost — APPROVED** [review]
@bvreede nice work. I could run the documentation locally. Just some minor comments, ready for a merge. :-)

**Pull request merged**